src/macaron/slsa_analyzer/checks/maven_repo_verification_check.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""A check to determine whether the source repository of a maven package can be independently verified."""
+
+import logging
+
+from packageurl import PackageURL
+from sqlalchemy import ForeignKey, Integer, String
+from sqlalchemy.orm import Mapped, mapped_column
+
+from macaron.database.table_definitions import CheckFacts
+from macaron.repo_finder.repo_finder_deps_dev import DepsDevRepoFinder
+from macaron.repo_verifier.repo_verifier_base import RepositoryVerificationStatus
+from macaron.slsa_analyzer.analyze_context import AnalyzeContext
+from macaron.slsa_analyzer.checks.base_check import BaseCheck
+from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence
+from macaron.slsa_analyzer.registry import registry
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class MavenRepoVerificationFacts(CheckFacts):
+    """The ORM mapping for justifications in maven source repo check."""
+
+    __tablename__ = "_maven_repo_verification_check"
+
+    #: The primary key.
+    id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
+
+    group: Mapped[str] = mapped_column(String, nullable=False)
+    artifact: Mapped[str] = mapped_column(String, nullable=False)
+    version: Mapped[str] = mapped_column(String, nullable=False)
+
+    # Repository link identified by Macaron's repo finder.
+    repo_link: Mapped[str] = mapped_column(String, nullable=True)
+
+    # Repository link identified by deps.dev.
+    deps_dev_repo_link: Mapped[str | None] = mapped_column(String, nullable=True)
+
+    # Number of stars on the repository identified by deps.dev.
+    deps_dev_stars_count: Mapped[int | None] = mapped_column(Integer, nullable=True)
```

Comment:
This property should be rendered as `TEXT` so, it's slightly different:
```suggestion
    deps_dev_stars_count: Mapped[int | None] = mapped_column(Integer, nullable=True, info={"justification": JustificationType.TEXT})
```